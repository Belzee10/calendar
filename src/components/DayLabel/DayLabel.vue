<template>
  <v-btn
    class="add-event"
    text
    small
    @mouseenter="handleHover"
    @mouseleave="handleHover"
    @click="addEvent"
  >
    {{ day }}
    <span v-if="showText" class="font-weight-light ml-1">Add event</span>
  </v-btn>
</template>

<script>
export default {
  name: 'DayLabel',
  props: {
    day: {
      type: Number,
      default: null
    },
    date: {
      type: String,
      default: ''
    }
  },
  data: () => ({
    showText: false
  }),
  methods: {
    handleHover() {
      this.showText = !this.showText;
    },
    addEvent() {
      this.$emit('add-event', {
        date: this.date
      });
    }
  }
};
</script>

<style lang="scss" scoped></style>
