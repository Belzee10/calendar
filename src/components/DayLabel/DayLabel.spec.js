import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import DayLabel from './DayLabel.vue';

const localVue = createLocalVue();

describe('DayLabel.vue', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  test('should show and hide "add event" text when mouseenter and mouseleave', () => {
    const wrapper = mount(DayLabel, {
      vuetify,
      localVue
    });

    wrapper.trigger('mouseenter');
    expect(wrapper.text()).toContain('Add event');
    wrapper.trigger('mouseleave');
    expect(wrapper.text()).not.toContain('Add event');
  });
});
