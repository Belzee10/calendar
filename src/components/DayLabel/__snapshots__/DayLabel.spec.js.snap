// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayLabel.vue should render correclty 1`] = `
<button
  class="add-event v-btn v-btn--flat v-btn--text theme--light v-size--small"
  type="button"
>
  <span
    class="v-btn__content"
  >
    
  1
  
    <!---->
  </span>
</button>
`;
